x/staking/module.go

```diff
@@ -47,8 +54,8 @@ func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
 
 // DefaultGenesis returns default genesis state as raw bytes for the staking
 // module.
-func (AppModuleBasic) DefaultGenesis() json.RawMessage {
-	return ModuleCdc.MustMarshalJSON(DefaultGenesisState())
+func (amb AppModuleBasic) DefaultGenesis() json.RawMessage {
```

Comment:
This initially looks like a code smell and `DefaultGenesis` turns into a misnomer because now you're not returning default module genesis state. You're returning pseudo-default genesis state with something that can be overridden. Also, how will this provide a solution to the issues laid out? How will `init` utilize this functionality? 

Since this annoyance comes from the deployment of rapid dev/testnets only, I'd again argue that this is not necessary and adds no real benefit. In your infrastructure, you should have a good understanding of what parameters you need to set for networks of all types. Simply have a step (e.g. simple bash or python script) that sets these parameters.